### PR TITLE
Deprecated CupertinoDialog in favor of a new widget called CupertinoP…

### PR DESCRIPTION
**Pull request by matthew-carroll**
Deprecated CupertinoDialog in favor of a new widget called CupertinoPopupSurface. (#20397)

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 1]
```diff
@@ -302,6 +248,87 @@ class CupertinoAlertDialog extends StatelessWidget {
...
+      ),
+    );
+  }
+}
+
+/// Rectangular surface that looks like an iOS popup surface, e.g., alert dialog
+/// and action sheet.
+///
+/// A [CupertinoPopupSurface] can be configured to paint or not paint its white
+/// foreground color. Typical usage should paint the foreground. The foreground
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/dialog.dart]: Clarify wording wrt 'foreground'. It's still the background from the perspective of its child. Specify that it's the thing between the blur and the child.

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 2]
```diff
@@ -302,6 +248,87 @@ class CupertinoAlertDialog extends StatelessWidget {
...
+/// can be disabled for the purpose of rendering divider gaps for more complicated
+/// layout, e.g., [CupertinoAlertDialog].
+///
+/// See also:
+///  * [CupertinoAlertDialog], which is a dialog with a title, content, and
+///    actions.
+///  * <https://developer.apple.com/ios/human-interface-guidelines/views/alerts/>
+class CupertinoPopupSurface extends StatelessWidget {
+  /// Creates an iOS-style rectangular popup surface.
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/dialog.dart]: rounded rectangle

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 3]
```diff
@@ -302,6 +248,87 @@ class CupertinoAlertDialog extends StatelessWidget {
...
+  const CupertinoPopupSurface({
+    Key key,
+    this.paintForeground = true,
+    this.child,
+  }) : super(key: key);
+
+  /// Whether or not to paint a translucent white foreground on top of this
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/dialog.dart]: Describe what it's for (i.e. why not let this always be false)

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 4]
```diff
@@ -302,6 +248,87 @@ class CupertinoAlertDialog extends StatelessWidget {
...
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return new ClipRRect(
+      borderRadius: BorderRadius.circular(_kDialogCornerRadius),
+      child: new BackdropFilter(
+        filter: new ImageFilter.blur(sigmaX: _kBlurAmount, sigmaY: _kBlurAmount),
+        child: new Container(
+          width: _kCupertinoDialogWidth,
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/dialog.dart]: This shouldn't be here right?

To test the general composability of this, move this to a different file and remove the word dialog from the various private consts that can now be moved.

**Comment by Hixie** [comment]
LGTM. This is probably the last time we will ever deprecate something, now that we are going into "stable" mode.

**Comment by matthew-carroll** [comment]
@Hixie "Ever"?

File: packages/flutter/lib/src/cupertino/dialog.dart (hunk 1 above)
Comment (matthew-carroll) [reply packages/flutter/lib/src/cupertino/dialog.dart]: done

File: packages/flutter/lib/src/cupertino/dialog.dart (hunk 2 above)
Comment (matthew-carroll) [reply packages/flutter/lib/src/cupertino/dialog.dart]: done

File: packages/flutter/lib/src/cupertino/dialog.dart (hunk 3 above)
Comment (matthew-carroll) [reply packages/flutter/lib/src/cupertino/dialog.dart]: the purpose is described in the class description

**Comment by Hixie** [comment]
Hopefully!

File: packages/flutter/lib/src/cupertino/dialog.dart (hunk 4 above)
Comment (matthew-carroll) [reply packages/flutter/lib/src/cupertino/dialog.dart]: I moved the width decision up a level.

**Comment by matthew-carroll** [comment]
@Hixie how would that work?  Every framework I've ever heard of has had many breaking changes across releases....let alone deprecations....how will Flutter be different?

**Comment by xster** [comment]
LGTM

**Pull request merged**